src/can-deploy.ts

```diff
@@ -87,7 +90,7 @@ export class CanDeploy {
     logger.info(
       `Asking broker at ${this.options.pactBroker} if it is possible to deploy`,
     );
-    const deferred = q.defer<string[]>();
+    const deferred = q.defer<any>();
```

Comment:
Should we declare a union type here - e.g. `string | CanIDeployResponse` type thing?

Reply:
That's a good point. Because we're just reading the json out of the broker cli, though, I'm not sure if we want to repeat the type (and keep it in sync with the ruby). I'm not very typescript savvy, so I'm not sure what the best practice would be.

Reply:
I would declare an interface that matches the response type, so that it can be navigated via VS code etc. even in vanilla JS. Otherwise it's just an `any` object.

I think it's OK to repeat the type info here, because if it changes in Ruby it's a breaking, backwards-incompatible change (so it shouldn't be done without a major version bump).

A union type is the way to go I think, and I think it should actually be `Error | CanDeployResponse`

Reply:
This is actually a bit more complex, I think.

The current behaviour in this PR is one of:

* Output = json, Can deploy, resolve with the object returned from the broker
* Output = table, Can deploy, resolve with the human readable output
* Output = json, Can't deploy, reject with the object returned from the broker
* Output = table, Can't deploy, reject with the human readable output.
* An error occurred, reject with the error

This means that the type of the promise is `CanDeployResponse | string`, and a rejected promise might be `CanDeployResponse | string | Error` (but since the rejecting type isn't specified anywhere, we don't have the `Error` explicitly anywhere).

I don't particularly like that the promise might reject with something that isn't an `Error`, but also I don't think we should resolve the promise in the case where the server says you can't deploy.

Maybe the right thing to do is to make a custom error type for failed can-i-deploy checks?

Reply:
> I don't particularly like that the promise might reject with something that isn't an Error, but also I don't think we should resolve the promise in the case where the server says you can't deploy.

You make a good point. I like the custom `Error` types idea - what are you thinking?

Reply:
Did we want to pull this change in as is, get in https://github.com/pact-foundation/pact-node/pull/190 and then update to use the Error interface before releasing? Feels like the two PRs are overlapping, and want to avoid a mega PR change.

It's definitely looking _better_, though.